ratatui-widgets/examples/sparkline.rs

```diff
@@ -24,29 +24,22 @@ use ratatui::{
     symbols,
     text::{Line, Span},
     widgets::{RenderDirection, Sparkline},
-    DefaultTerminal, Frame,
+    Frame,
 };
 
 fn main() -> Result<()> {
     color_eyre::install()?;
-    let terminal = ratatui::init();
-    let result = run(terminal);
-    ratatui::restore();
-    result
-}
-
-/// Run the application.
-fn run(mut terminal: DefaultTerminal) -> Result<()> {
-    loop {
-        terminal.draw(draw)?;
-        if event::poll(Duration::from_millis(16))? && matches!(event::read()?, Event::Key(_)) {
+    let frame_timeout = Duration::from_secs_f64(1.0 / 60.0); // run at 60 FPS
+    ratatui::run(|terminal| loop {
+        terminal.draw(render)?;
+        if event::poll(frame_timeout)? && matches!(event::read()?, Event::Key(_)) {
```

Comment:
```suggestion
        if event::poll(frame_timeout)? && matches!(event::read()?, Event::Key(x) if x.kind == KeyEventKind::Press ) {
```

Reply:
For the most part, because this immediately exits, I'm not too concerned about checking for whether it's a key press or release. Currently unreleased in crossterm there is code that adds an `Event::is_key_press()` method, so these checks will become e.g. `if event::poll(frame_timeout)? && event::read()?.is_key_press() {` when that is released.

The only real concern that really require this is that I have heard it rumored that sometimes windows terminals will pass through the key release event of the enter key that was used to run the command that runs the app. I don't use windows often enough to have seen this personally though, so it's a hypothetical problem right now.

Reply:
> The only real concern that really require this is that I have heard it rumored that sometimes windows terminals will pass through the key release event of the enter key that was used to run the command that runs the app. I don't use windows often enough to have seen this personally though, so it's a hypothetical problem right now.

I've seen this actually happen a few times since then, but the change would be to check `event::read()?.is_key_press()` which was introduced in crossterm 0.29 instead.